src/init.cpp

```diff
@@ -1348,13 +1348,20 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         onion_proxy = addrProxy;
     }
 
+    const bool onlynet_used_with_onion{args.IsArgSet("-onlynet") && IsReachable(NET_ONION)};
+
     // -onion can be used to set only a proxy for .onion, or override normal proxy for .onion addresses
     // -noonion (or -onion=0) disables connecting to .onion entirely
     // An empty string is used to not override the onion proxy (in which case it defaults to -proxy set above, or none)
     std::string onionArg = args.GetArg("-onion", "");
     if (onionArg != "") {
         if (onionArg == "0") { // Handle -noonion/-onion=0
             onion_proxy = Proxy{};
+            if (onlynet_used_with_onion) {
+                return InitError(
+                    _("Outbound connections restricted to Tor (-onlynet=onion) but the proxy for "
+                      "reaching the Tor network is explicitly forbidden: -onion=0"));
```

Comment:
nit, use the same (more explicit) phrasing here as for the error message below, and consistent verb tense
```diff
-                    _("Outbound connections restricted to Tor (-onlynet=onion) but the proxy for "
-                      "reaching the Tor network is explicitly forbidden: -onion=0"));
+                    _("The -onlynet=onion configuration option was passed to restrict outbound"
+                      " connections to Tor, but the proxy for reaching the Tor network was explicitly forbidden: -onion=0"));
```

Reply:
Would be good to have consistent wording here and for I2P and CJDNS in https://github.com/bitcoin/bitcoin/pull/25989.

`master`:

> Outbound connections restricted to Tor (-onlynet=onion) but the proxy for  reaching the Tor network is not provided ...

this PR:

> Outbound connections restricted to Tor (-onlynet=onion) but the proxy for reaching the Tor network is explicitly forbidden: -onion=0

> The -onlynet=onion configuration option was passed to restrict outbound connections to Tor, but the proxy for reaching the Tor network was not provided ...

https://github.com/bitcoin/bitcoin/pull/25989

> Outbound connections restricted to CJDNS (-onlynet=cjdns) but -cjdnsreachable is not provided

> Outbound connections restricted to i2p (-onlynet=i2p) but -i2psam is not provided

Reply:
I think the explicit version is clearer.